internal/pkg/routes/route.go

```diff
@@ -60,10 +60,14 @@ func (r route) Process(data []byte) ([]byte, error) {
 }
 
 func (r route) Consume(msg *nats.Msg) {
-	payload, err := r.Process(msg.Data)
-	if err != nil {
-		r.logger.Error(fmt.Sprintf("Failed to consume msg %s", err.Error()))
-	}
-	topic := fmt.Sprintf("%s/%s", r.MqttTopic(), strings.ReplaceAll(msg.Subject, ".", "/"))
-	r.pub.Publish(msg.Subject, topic, payload)
+	go func() {
```

Comment:
I am not sure that it is good to have the whole function which just calls a goroutine. I would prefer that this fuction does not call a goroutine, but the function itself is called as `go Consume()`. This way we would be capable to call it with or without concurrence.

I am not sure that this is a correct approach, and would like to hear opinions from @dusanb94 @nmarcetic @blokovi and @darkodraskovic

Reply:
@drasko I agree. That's the recommended way and we should stick with it.